core/src/plugins/container/cloudbuilder.ts

```diff
@@ -359,6 +360,63 @@ class CloudBuilder {
 
 export const cloudBuilder = new CloudBuilder()
 
+function isContainerBuilderEnabled({
+  ctx,
+  containerProviderConfig,
+}: {
+  ctx: PluginContext
+  containerProviderConfig: ContainerProviderConfig
+}) {
+  const apiVersion = ctx.projectApiVersion
+
+  if (containerProviderConfig.gardenCloudBuilder !== undefined) {
+    reportDeprecatedFeatureUsage({ apiVersion, log: ctx.log, deprecation: "gardenCloudBuilder" })
+  }
+
+  if (gardenEnv.GARDEN_CLOUD_BUILDER !== undefined) {
+    reportDeprecatedFeatureUsage({ apiVersion, log: ctx.log, deprecation: "gardenCloudBuilderEnvVar" })
+  }
+
+  if (!!containerProviderConfig.gardenContainerBuilder && !!containerProviderConfig.gardenCloudBuilder) {
+    throw new ConfigurationError({
+      message: deline`
+      Provider configuration declares both ${styles.highlight("gardenContainerBuilder")} and ${styles.highlight("gardenCloudBuilder")} fields.
+      Please use only ${styles.highlight("gardenContainerBuilder")}.
+      `,
+    })
+  }
+
+  // handle new config
+  if (!!containerProviderConfig.gardenContainerBuilder) {
+    let isCloudBuilderEnabled = containerProviderConfig.gardenContainerBuilder.enabled || false
+
+    // The env variable GARDEN_CONTAINER_BUILDER can be used to override the gardenContainerBuilder.enabled config setting.
+    // It will be undefined, if the variable is not set and true/false if GARDEN_CONTAINER_BUILDER=1 or GARDEN_CONTAINER_BUILDER=0.
+    const overrideFromEnv = gardenEnv.GARDEN_CONTAINER_BUILDER || gardenEnv.GARDEN_CLOUD_BUILDER
+    if (overrideFromEnv !== undefined) {
+      isCloudBuilderEnabled = overrideFromEnv
+    }
+
+    return isCloudBuilderEnabled
+  }
+
+  // handle old config
+  if (!!containerProviderConfig.gardenCloudBuilder) {
+    let isCloudBuilderEnabled = containerProviderConfig.gardenCloudBuilder.enabled || false
+
+    // The env variable GARDEN_CLOUD_BUILDER can be used to override the gardenCloudBuilder.enabled config setting.
+    // It will be undefined, if the variable is not set and true/false if GARDEN_CLOUD_BUILDER=1 or GARDEN_CLOUD_BUILDER=0.
+    const overrideFromEnv = gardenEnv.GARDEN_CLOUD_BUILDER
```

Comment:
Here we also need to take into account both env vars, as the error message advertises only `GARDEN_CONTAINER_BUILDER` as an escape hatch.
```suggestion
    const overrideFromEnv = gardenEnv.GARDEN_CONTAINER_BUILDER || gardenEnv.GARDEN_CLOUD_BUILDER
```